### ISSUE TITLE
Feature Request: Option to unify column layout for both Downloads and Uploads tabs

### ISSUE THREAD

**Issue by redactedscribe** [Nicotine-Plus/nicotine-plus#980]
### Describe the solution you'd like
Downloads and Uploads could have the option to have their columns mirrored, i.e. "Upload mode's columns mirrors that of Downloads'" / "Upload and Downloads share the same column layout", (the former would be more self-evident as to how I see it actually working). I can see why people would like different settings for each, but it could also be nice to have only one "setting" (arrangement) for both so they stay in sync, which could be a better experience by default.

`Nicotine+ v3.0.0.dev1`

**Comment by alekksander** [comment]
i cannot agree. having width permanently correlated could cause view problems as upload and download columns information is not equally important. as well this could be painful on smaller resolutions. whats more, some of the columns are not necessary on both views in certain conditions (like queue on upload tab if there are no limits setup on upload transfer), time elapsed/remaining is also a column with nice, but not that crucial information sometimes. having width customized manually is not that much of work. eventually solution like RMB on column → „set this width globally” is okay i think without causing problems. dunno if necessary though, but luckily it's not my decision to make

**Comment by redactedscribe** [comment]
I agree that columns should be independently customizable for the upload and download views (as it is currently), I do not want to change this. The columns in both views are already identical out-of-the-box and I do not think it presents much of an issue as you say it does. The feature request was a suggestion for an _option_ so that information on both views doesn't "jump" around (left/right). I like your idea for a "set width for upload/download view" context menu option, as it would still allow for columns to be mirrored, though not automatically (which is essentially what I was asking for).

**Comment by mathiascode** [comment]
This request is blocked until we've ported the transfer list to the new TreeView class.
